staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -37,6 +40,9 @@ const (
 	// auditKey is the context key for storing the audit event that is being
 	// captured and the evaluated policy that applies to the given request.
 	auditKey
+
+	// auditAnnotationsMutexKey is the context key for the audit annotations mutex.
+	auditAnnotationsMutexKey
```

Comment:
Why does it have to be two separate keys rather than just store a `*struct{Mutex, []annotation}`?

Reply:
In case you ever have the audit event registered without the annotations... this whole thing is a mess. I'm planning on refactoring this to consolidate it all into a single audit context / key, but that's the larger change that I'm holding back for v1.25.

Reply:
Basically this test case: https://github.com/kubernetes/kubernetes/blob/abebdbaa8aba0b774e6cf0cccfc0e50797776265/staging/src/k8s.io/apiserver/pkg/audit/context_test.go#L66-L68

Reply:
That makes me even more confused, isn't that an especially good reason to not have two keys?

Reply:
I see, there's actually already two keys and a TODO to combine them. This change takes us up to three keys.

Reply:
Yeah. I'm happy to condense them, I just think it will make the change a bit bigger.

Reply:
I mean, on the one hand, I don't want you to have to do extra work at this late hour; on the other, how do we know no one is accessing the map in a non standard way? Hm, I guess the unexported key type probably helps.

Reply:
I don't really understand how this code got like this, but I think this is an improvement :/

I would like us to change it to have just one key in the future though. Can you extend the TODO above to also mention folding in the mutex, too?

Reply:
Updated the TODO.